### ISSUE TITLE
Mypy is not inferring `__all__` in `__all__=[]` to be `list[str]`

### ISSUE THREAD

**Issue by denballakh** [python/mypy#13997]
```py
__all__ = [] # error: Need type annotation for "__all__" (hint: "__all__: List[<type>] = ...")
```

**Expected Behavior**
No errors. `__all__` is inferred to be `list[str]`

**Actual Behavior**
Mypy is complaining about untyped empty list

I think there are simiar problems with other dunders, which type is known statically.
```py
__all__ = [] # error
class X:
    __slots__ = [] # error
    __match_args__ = [] # error, but it is ok, because it must be tuple

__annotations__ = {} # no error :)
```

**Your Environment**
- Mypy version used: mypy 0.982 (compiled: no)
- Mypy command-line flags: no
- Mypy configuration options from `mypy.ini` (and other config files): no
- Python version used: 3.11.0

**Comment by randolf-scholz** [comment]
duplicate: https://github.com/python/mypy/issues/10103
